spec/sidekiq/text_extraction_worker_spec.rb

```diff
@@ -9,40 +9,92 @@
  
   describe "#perform" do
     context 'when the PDF extraction is not part of a harvest' do
-      before do
-        FileUtils.cp("#{Rails.root}/spec/support/example.pdf", "#{extraction_job.extraction_folder}/example__1234__01.json")
-      end
 
-      it 'converts a PDF into raw text' do
-        extracted_files = Dir.glob("#{extraction_job.extraction_folder}/*").select { |e| File.file? e }
+      context 'when the PDF has a text layer' do
+        before do
+          FileUtils.cp("#{Rails.root}/spec/support/example.pdf", "#{extraction_job.extraction_folder}/example__1234__01.json")
+        end
+  
+        it 'converts a PDF into raw text' do
+          extracted_files = Dir.glob("#{extraction_job.extraction_folder}/*").select { |e| File.file? e }
+    
+          expect(extracted_files.count).to eq 1
+  
+          TextExtractionWorker.new.perform(extraction_job.id)
   
-        expect(extracted_files.count).to eq 1
+          extracted_files = Dir.glob("#{extraction_job.extraction_folder}/*").select { |e| File.file? e }
```

Comment:
question: probably a fundamental ruby question but do you need to redefine the variable in order to reevaluate this? Not sure if this is a pass by reference vs pass by value question